test/nuxeo-filter.html

```diff
@@ -321,6 +382,58 @@
       removalContainer.removeChild(filter);
       expect(stamped(removalContainer, '.toRemove').length).to.be.equal(0);
     });
+
+    /* global customElementsAndBehaviorsContainer */
+    test('filter custom elements', () => {
+      fixture('customElementsAndBehaviorsContainer');
+      document.getElementById('mySlotContent').create();
+      return flush().then(() => {
+        const stampedElements = stamped(customElementsAndBehaviorsContainer, '.custom');
+        expect(stampedElements.length).to.be.equal(1);
+        let myElement = stampedElements[0];
+        expect(stampedOnShadowRoot(myElement, '#label').length).to.be.equal(1);
+        expect(myElement.label).to.be.equal('A simple test');
+        const slot = stamped(customElementsAndBehaviorsContainer, 'nuxeo-slot')[0];
+        slot.set('model', {document: {type: 'File'}, text: 'A simple test'});
+        return flush().then(() => {
```

Comment:
You should be able to just return the flush here and do the `then` one level up, thus preventing promise nesting.